### DISCUSSION TITLE
Integrating PyQt5 in Flet UI

### DISCUSSION THREAD

**Discussion by Amir4python** [flet-dev/flet#2184] [General]
`

def main(page: ft.Page):
    # DEFINE UTILITY buttons

    page.title = "Something1 and something2"

   ###### SOME INTERMEDIATE CODE IS THERE ####
  secondViewTitle = ft.AppBar(title=ft.Text("do_something2"), bgcolor=ft.colors.YELLOW_400, color=ft.colors.BLACK, toolbar_height=35)
  
      secondView = ft.View(route="/do_something2", controls=[secondViewTitle, MainButton])
  
      if page.route == "/do_something2":
     
          page.views.append(secondView)
  
          pyqt_demo.main() # this is PyQt5 main function`

> pyqt_demo.main() is entry point function that open pyQt GUI app.
> When i run the program  , pyqt window application opens up, The flet UI's main page becomes completely blank.  The main UI has two buttons that routes to the view pages (dosomething1 for first View and dosomething for second View). Do dosomething1 is completely a flet python code , but that is not an isssue, the main issue is with the do_something2 that calls the main function of Pyqt5 app package code.
> i would get ---> warning: qapplication was not created in the main() thread. when running flet main page code

**Comment by mouadessalim** [comment]
Hey there,

Thanks for opening the GitHub issue. We're always eager to receive contributions and feedback. However, it seems that the issue description you provided is a bit unclear and doesn't align with our project's conventions.

To maintain a structured and efficient collaboration environment, we insist on adherence to the rules set forth in our `CONTRIBUTING.md` file. It's our go-to resource for best practices and guidelines for issue creation and resolution.

GitHub is where we prioritize structured, focused contributions. It's not a casual chat room. It's essential that your issues are well-structured to keep the project organized and productive.

Here are a couple of developer-friendly tips:

1. Please review our `CONTRIBUTING.md` document: It's like our project's rulebook. Take some time to dive into it to ensure your issue follows our project's guidelines. It'll help you provide a clear, meaningful contribution.

2. Revise your issue description: We'd appreciate it if you could fine-tune your GitHub issue to make it more comprehensible and in line with our project's guidelines. Don't forget to include all relevant details, like steps to reproduce the issue, expected and actual behavior, and any other necessary information.

By following these guidelines, you'll be helping us maintain a well-oiled project machine. Your contributions are important, and these adjustments will benefit not just your interaction but the entire community.

Thanks for your understanding, and we look forward to your revised issue.

**Comment by ndonkoHenri** [comment]
@Amir4python please have a look [here](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks#fenced-code-blocks) on how to format your code when sharing it on GitHub.

Also, make sure you share a repro code of your issue so we can test it out and if possible, provide assistance.

**Comment by Amir4python** [comment]
https://github.com/Amir4python/MultiApps  
You can this test, run it

**Comment by Amir4python** [comment]
@mouadessalim  Just test this repo , u will understand it.
https://github.com/Amir4python/MultiApps
